libs/common-services/src/main/java/uk/gov/di/ipv/core/library/persistence/InMemoryDataStore.java

```diff
@@ -0,0 +1,205 @@
+package uk.gov.di.ipv.core.library.persistence;
+
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbPartitionKey;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbSecondaryPartitionKey;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbSortKey;
+import uk.gov.di.ipv.core.library.config.ConfigurationVariable;
+import uk.gov.di.ipv.core.library.exceptions.ItemAlreadyExistsException;
+import uk.gov.di.ipv.core.library.persistence.item.PersistenceItem;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+public class InMemoryDataStore<T extends PersistenceItem> implements DataStore<T> {
+    private static final String KEY_SEPARATOR = "/";
+    private static final ConcurrentMap<String, Map<String, ?>> TABLES = new ConcurrentHashMap<>();
+
+    private final ConcurrentMap<String, T> records;
+    private final Class<T> klass;
+    private final Method partitionKeyMethod;
+    private final Method sortKeyMethod;
+    private final boolean hasSortKey;
+
+    @SuppressWarnings("unchecked") // Rely on callers to use consistent types per table
+    public InMemoryDataStore(String tableName, Class<T> klass) {
+        Method partitionKey = null;
+        Method sortKey = null;
```

Comment:
Do you need these intermediate vars? Can you just assign to the fields in the for loop?

Reply:
If you don't use the intermediate vars then you'd need to initialise the instance variables to null and make them non-final, which feels a bit unsatisfactory.

We could extract the method search to avoid it though...